### ISSUE TITLE
use generator in getStatesFromFasta 

### ISSUE THREAD

**Issue by philippmuench** [hiddengenome/deepG#63]
- currently each fasta file is preprocessed and held in RAM, preprocessing should happen within the generator, like we are doing during training
- replace predict with predict generator https://keras.rstudio.com/reference/predict_generator.html

**Comment by mr-francois** [comment]
`predict_generator` is slightly faster than `predict_on_batch` (~90% run time) but would be problematic with other features:
1) For padding we would have to switch between  `predict_generator` and `predict_on_batch` 
2) Difficult to write states to h5 in batches while keeping track of current position in sequence
